libsolidity/interface/CompilerStack.cpp

```diff
@@ -1285,12 +1285,30 @@ void CompilerStack::generateEVMFromIR(ContractDefinition const& _contract)
 	//cout << yul::AsmPrinter{}(*stack.parserResult()->code) << endl;
 
 	// TODO: support passing metadata
-	auto result = stack.assemble(yul::AssemblyStack::Machine::EVM);
-	compiledContract.object = std::move(*result.bytecode);
-	// TODO: support runtimeObject
-	// TODO: add EIP-170 size check for runtimeObject
+	// TODO: use stack.assemble here!
+	yul::MachineAssemblyObject init;
+	yul::MachineAssemblyObject runtime;
+	std::tie(init, runtime) = stack.assembleAndGuessRuntime();
```

Comment:
Not necessarily happy with using this, but this is a simple enough way for now to support `runtimeObject` -- unless we can agree to remove that in the long term and no need for it here.

Reply:
Because using `stack.assemble` would require a lot more work? Or what's the reason?

Reply:
Actually, I guess it can be done. What we need is extracting the runtime object from `compiledContract.yulIROptimized` and try compiling that with `stack.assemble()`.

The logic `assembleAndGuessRuntime` does is if there's a single subobject, then that is the runtime, so we can follow the same here.

Reply:
Ok. Do you wanna do that in this PR or keep the TODO and open an issue about it?

Reply:
I actually tried this locally, but it was messy. For the sake of progress I'd just merge this, and then in a next step try to use the other method and also export `Assembly`. That would enable most of the features, since #10348 was merged.

Reply:
We know the name of the runtime and should extract it by its name.